### PR TITLE
pki: T3970: Allow op-mode PKI commands in a config session to install directly

### PR DESCRIPTION
**Pull request by sarthurdev**
## Change Summary

Allows existing PKI op-mode `install <name>` syntax to install directly when called from configure mode

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T3970

## Component(s) name

pki

## Proposed changes

- Adds function `vyos.util.install_into_config` to show or install config values depending if a config session is available.
  - Has user prompt to ensure no existing config values are overwritten without user consent
- Adds all PKI `install <name>` op-mode commands to use `install_into_config` allowing for directly installing PKI values into pending config of active config session.

## How to test

Op-mode:
```
vyos@vyos:~$ generate pki ca install test
Enter private key type: [rsa, dsa, ec] (Default: rsa) ec
...
You are not in configure mode, commands to install manually from configure mode:
set pki ca test certificate '...'
set pki ca test private key '...'
```

Conf-mode:
```
vyos@vyos:~$ conf
WARNING: You are currently configuring a live-ISO environment, changes will not persist until installed
[edit]
vyos@vyos# run generate pki ca install test
Enter private key type: [rsa, dsa, ec] (Default: rsa) ec
...
2 value(s) installed. Use "compare" to see the pending changes, and "commit" to apply.
[edit]
vyos@vyos# comp
+pki {
+    ca test {
+        certificate ...
+        private {
+            key ...
+        }
+    }
+}
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: python/vyos/util.py [hunk 1]
```diff
@@ -903,3 +902,34 @@ def check_port_availability(ipaddress, port, protocol):
...
+    # override_prompt: if True, user will be prompted before existing nodes are overwritten
+
+    if not config_paths:
+        return None
+
+    from vyos.config import Config
+
+    conf = Config()
+
+    if not conf.in_session():
```
Comment (c-po) [inline python/vyos/util.py]: Can we use `Config().in_session()` here?

File: python/vyos/util.py [hunk 2]
```diff
@@ -903,3 +902,34 @@ def check_port_availability(ipaddress, port, protocol):
...
+        print('You are not in configure mode, commands to install manually from configure mode:')
+        for path in config_paths:
+            print(f'set {path}')
+        return None
+
+    count = 0
+
+    for path in config_paths:
+        if override_prompt and conf.exists(path) and not conf.is_multi(path):
```
Comment (c-po) [inline python/vyos/util.py:925]: Can we use an instance of `ConfigTreeQuery` for `conf`?

**Review by c-po — CHANGES_REQUESTED** [review]

File: python/vyos/util.py (hunk 2 above)
Comment (sarthurdev) [reply python/vyos/util.py:925]: As the underlying code essentially uses the `Config` class and we needed an instance for the session check, it seemed more efficient to use just the single instance for both checks.

Comment (c-po) [reply python/vyos/util.py:925]: Please still use ConfigTreeQuery whenever possible from op mode.

**Pull request merged**